hpOneView/resources/settings/appliance_device_snmp_v1_trap_destinations.py

```diff
@@ -19,51 +19,91 @@
 from __future__ import division
 from __future__ import print_function
 from __future__ import unicode_literals
-
 from future import standard_library
 
 standard_library.install_aliases()
 
-from hpOneView.resources.resource import ResourceClient
+
+from hpOneView.resources.resource import Resource
 
 
-class ApplianceDeviceSNMPv1TrapDestinations(object):
+class ApplianceDeviceSNMPv1TrapDestinations(Resource):
     """
     ApplianceDeviceSNMPv1TrapDestinations API client.
-    The appliance has the ability to forward events received from monitored or managed server hardware to the specified destinations as SNMPv1 traps.
+
+    The appliance has the ability to forward events received from monitored or managed
+    server hardware to the specified destinations as SNMPv1 traps.
     """
     URI = '/rest/appliance/trap-destinations'
 
-    def __init__(self, con):
-        self._client = ResourceClient(con, self.URI)
+    def __init__(self, connection, data=None):
+        super(ApplianceDeviceSNMPv1TrapDestinations, self).__init__(connection, data)
 
-    def create(self, resource, id=None, timeout=-1):
+    def get_by_id(self, id):
+        """
+        Retrieves the trap destination based on the id specified.
+
+        Args:
+            id (int): id of the snmp_v1 trap resource
+
+        Return:
+            dict: SNMPv1 trap destination
+
+        """
+        uri = "{0}/{1}".format(self.URI, id)
+        return self._helper.do_get(uri)
+
+    def create_validation(self, data=None, timeout=-1):
```

Comment:
Request body of the validation endpoint having only four fields. 
Instead of accepting a dictionary for these fields, can we add four arguments for this method and document those fields in the docstring?